Common/TableProducer/qVectorsTable.cxx

```diff
@@ -444,79 +405,124 @@ struct qVectorsTable {
+                              objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 1, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 2, i + 1));
+
+          helperEP.DoRecenter(qvecRe[(kBNeg + 1) * 4 * id + i * 4 + 2], qvecIm[(kBNeg + 1) * 4 * id + i * 4 + 2],
+                              objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 1, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 2, i + 1));
+          helperEP.DoTwist(qvecRe[(kBNeg + 1) * 4 * id + i * 4 + 2], qvecIm[(kBNeg + 1) * 4 * id + i * 4 + 2],
+                           objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 3, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 4, i + 1));
+
+          helperEP.DoRecenter(qvecRe[(kBNeg + 1) * 4 * id + i * 4 + 3], qvecIm[(kBNeg + 1) * 4 * id + i * 4 + 3],
+                              objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 1, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 2, i + 1));
+          helperEP.DoTwist(qvecRe[(kBNeg + 1) * 4 * id + i * 4 + 3], qvecIm[(kBNeg + 1) * 4 * id + i * 4 + 3],
+                           objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 3, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 4, i + 1));
+          helperEP.DoRescale(qvecRe[(kBNeg + 1) * 4 * id + i * 4 + 3], qvecIm[(kBNeg + 1) * 4 * id + i * 4 + 3],
+                             objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 5, i + 1), objQvec.at(id)->GetBinContent(static_cast<int>(cent) + 1, 6, i + 1));
+        }
       }
+      qvecReFT0C.push_back(qvecRe[(kBNeg + 1) * 4 * id + kFT0C * 4 + 3]);
+      qvecImFT0C.push_back(qvecIm[(kBNeg + 1) * 4 * id + kFT0C * 4 + 3]);
+      qvecReFT0A.push_back(qvecRe[(kBNeg + 1) * 4 * id + kFT0A * 4 + 3]);
+      qvecImFT0A.push_back(qvecIm[(kBNeg + 1) * 4 * id + kFT0A * 4 + 3]);
+      qvecReFT0M.push_back(qvecRe[(kBNeg + 1) * 4 * id + kFT0M * 4 + 3]);
+      qvecImFT0M.push_back(qvecIm[(kBNeg + 1) * 4 * id + kFT0M * 4 + 3]);
+      qvecReFV0A.push_back(qvecRe[(kBNeg + 1) * 4 * id + kFV0A * 4 + 3]);
+      qvecImFV0A.push_back(qvecIm[(kBNeg + 1) * 4 * id + kFV0A * 4 + 3]);
+      qvecReBPos.push_back(qvecRe[(kBNeg + 1) * 4 * id + kBPos * 4 + 3]);
+      qvecImBPos.push_back(qvecIm[(kBNeg + 1) * 4 * id + kBPos * 4 + 3]);
+      qvecReBNeg.push_back(qvecRe[(kBNeg + 1) * 4 * id + kBNeg * 4 + 3]);
+      qvecImBNeg.push_back(qvecIm[(kBNeg + 1) * 4 * id + kBNeg * 4 + 3]);
     }
 
-    // Fill the columns of the Qvectors table if they are found for a detector.
+    // Fill the columns of the Qvectors table.
     qVector(cent, IsCalibrated, qvecRe, qvecIm, qvecAmp);
     if (useDetector["QvectorFT0Cs"])
-      qVectorFT0C(IsCalibrated, qvecRe[kFT0C * 4 + 3], qvecIm[kFT0C * 4 + 3], sumAmplFT0C);
+      qVectorFT0C(IsCalibrated, qvecReFT0C, qvecImFT0C, qvecAmp[kFT0C]);
```

Comment:
Dear @jikim1290,
I looked carefully at your PR and this is not the change we discussed offline.
I'm taking as an example the qVectorFT0C table. Here you are replacing qvecRe[kFT0C *4 +3], which is the Q-vector at the last step of correction, with qvecReFT0C, which is the array with Q vectors at each correction step.
However, the proposal was simply to have the possibility to configure the correction step that an analyser can use. 
So our proposal was to replace  `qvecRe[kFT0C * 4 + 3], qvecIm[kFT0C * 4 + 3]` with `qvecRe[kFT0C * 4 + cfgCorrStep], qvecIm[kFT0C * 4 + cfgCorrStep]`. In this way, if an analyser set cfgCorrStep == 3 gets the fully-corrected Q vector, while if cfgCorrStep == 1 gets the Q vector with only Recentering correction applied.

Is there a reason for the current PR where you are storing everything in the detector-specific table of Q vectors?